client/my-sites/themes/recommended-themes.jsx

```diff
@@ -37,12 +37,14 @@ class RecommendedThemes extends Component {
 }
 
 const ConnectedRecommendedThemes = connect(
-	( state ) => {
+	( state, props ) => {
+		const { isLoadingCoreFSESettings } = props;
 		const siteId = getSelectedSiteId( state );
-		const filter = getRecommendedThemesFilter( state, siteId );
+		const filter = getRecommendedThemesFilter( siteId );
+
 		return {
 			customizedThemesList: getRecommendedThemesSelector( state, filter ),
-			isLoading: areRecommendedThemesLoading( state, filter ),
+			isLoading: isLoadingCoreFSESettings || areRecommendedThemesLoading( state, filter ),
```

Comment:
Wait for the Gutenberg FSE Settings to arrive before showing the themes. Not having this resulted in a flash of stale data being shown.

Reply:
The fact that you need a query hook and a selector to get the query result is suspicious. That's the Redux way to have them separate. With React Query, you should need just one hook call and use its return value:
```js
const { isLoading, data } = useGutenbergFSESettings( siteId );
/* work with isLoading and data */
getRecommendedThemesFilter( siteId, data?.is_core_fse_eligible );
```
You'll need to rework the `ConnectedRecommendedThemes` to use the `useGutenbergFSESettings` hook (and not use it anywhere else) and combine the data from React Query and Redux to construct all the props to be passed to the `<RecommendedThemes>` component.